### DISCUSSION TITLE
Access text of another element and pass it to some function

### DISCUSSION THREAD

**Discussion by florian-sun** [alpinejs/alpine#4465] [1. Help] [answered]
How can I get the text of another element and use it by passing to a function in current element?
Something like this (doesn't work):

```
<span  @click= "someFunction( '$refs.text' )">
  <span x-text="item" x-ref="text"></span>
</span>

```

**Comment by SimoTod** [comment]
you don't need to use refs to access internal data 
```html
<span @click= "someFunction(item)">
  <span x-text="item"></span>
</span>
```

in general $refs.text (without single quotes), return the html element so you need the browser api to get the text content from there (textContent or innerText)

**Reply by florian-sun** [reply]
Thanks but the actual code is a bit more complex, I don't think I can directly access internal data. 

```
<template x-for="item in items" :key="item"> 
  <span @click="someFunction(item)">
    <span x-text="item"> </span>
  </span>
</template>

```
I guess I have to use browser api.

**Reply by SimoTod** [reply, marked as answer]
have you tried?

https://codepen.io/SimoTod/pen/VYZvbQQ

**Comment by florian-sun** [comment]
I did try but I was using a global function, not a function defined in x-data.
It works, thanks a lot
